doc-deploy-dev/action.yml

```diff
@@ -134,9 +134,42 @@ runs:
       uses: actions/checkout@v4
       with:
         repository: ${{ env.REPOSITORY }}
-        ref: ${{ inputs.branch }}
         token: ${{ inputs.token }}
 
+    - name: "Check if branch exists"
+      env:
+        BRANCH: ${{ inputs.branch }}
+      shell: bash
+      run: |
+        # Check the gh-pages branch exists on remote
+        branch_exists=$(git ls-remote --heads origin refs/heads/${{ env.BRANCH }} 2>&1)
+
+        # If the ${{ env.BRANCH }} doesn't exist, then print error message and exit 1
+        if [ -z "$branch_exists" ]; then
+          echo "The ${{ env.BRANCH }} branch does not exist. Creating ${{ env.BRANCH }}."
+
+          # Create orphan branch
+          git checkout --orphan ${{ env.BRANCH }}
+
+          # Remove all files in staging area
+          git rm -rf .
```

Comment:
Shouldn't this be?

```suggestion
          git restore --staged .
```

Reply:
`git restore --staged .` didn't work for me after making an orphan branch (see pic below), but I could do `git rm --cached -r .` instead. It will just leave behind the untracked files from the repo instead of removing them with `git rm -rf .`

![rmcached](https://github.com/ansys/actions/assets/58492561/54816304-cb3f-4898-871b-b5a9927c7306)

Reply:
Sounds good